packages/core/docs/performance/good-practices.md

```diff
@@ -0,0 +1,344 @@
+
+Only images above the fold should be loaded on page render, rest should be lazy loaded.
+Remember don't lazy load image that is potential candidate for Largest Contentful Paint.
+
+::: tip HINTS
+- If you are using SFUI components eg. SfImage, all images are lazy loaded automatically. But if you are using your own implementation remember to use 
+```html
+<img loading=lazy>
+```
+ or eg. nuxt-lazy-load library (depending on your needs and browser support).
+- If you want to use lazy loading in Nuxt 2 go with this [vue-lazyload module](https://www.npmjs.com/package/vue-lazyload).
+- If you want to use lazy loading in Nuxt 3 go with this [lazy-load module](https://github.com/nuxt-modules/lazy-load).
+:::
+
+### Favicon should be small and cacheable ⚫⚪⚪
+
+It is easy to make the favicon big but please avoid doing that, because every browser will then perform an unnecessarily large download. And make sure the cache headers are set for a long time for the favicon. It is easy to miss since it's another content type.
+
+### Avoid scaling images in the browser ⚫⚫⚫
+
+If you need a 300x300 image (e.g. on mobile) don't download 1000x1000 and scale it in the browser, doing that is bad for performance. Users will download extra kilobytes of data that could be avoided. Also extra CPU will be needed to handle it. 
+
+::: tip HINTS
+- Make sure you create multiple versions of the same image server-side and serve the appropriate one depending on needed size (eg different on mobile and desktop).
+- In vue/nuxt applications you can use the nux-image library that will help you to resize and optimize images. 
+- You can use picture and srcset to load images depending on device resolution.
+:::
+## Other optimizations
+
+### Avoid render blocking resources ⚫⚫⚫
+
+Scripts and stylesheets can block the first paint of your page (if they are in head and don’t have async or defer for scripts and media or disabled for stylesheets).
+
+Consider delivering critical JS/CSS inline and deferring all non-critical JS/styles. You can reduce the size of your pages by only shipping the code and styles that you need.
+
+::: tip HINTS
+- Once you've identified critical code, move that code from the render-blocking URL to an inline script tag in your HTML page.
+- Inline critical styles required for the first paint inside a style block at the head of the HTML page and load the rest of the styles asynchronously using the preload link.
+:::
+
+### Total page size shouldn't be too big ⚫⚫⚫
+
+Avoid having pages that have a transfer size more than 2 MB (desktop) and 1 MB (mobile) because it will hurt performance and will make the page expensive for the user if she/he pays for the bandwidth.
```

Comment:
maybe recommend infinite scroll as alternative

Reply:
For now I'm not sure it's good option for perf, this needs to be checked, we will add it later if it helps.

Reply:
From some point of view this document might be also a guide for client / dev to give him alternative solutions for common problems (not everywhere applicable ofc, but it would be nice addition, imo), e.g. when need to show all products on one page (common requirement these days) and is unfamiliar with this concept, to say to him: _"Hey, its still possible without loading all your product in 1 shot and hurt performance"_. 
Performance doesn't always involve trade-offs.